src/Datadog.Trace/Logging/LogRateLimiter.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Concurrent;
+
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.Logging
+{
+    internal class LogRateLimiter : ILogRateLimiter
+    {
+#if NET45
+        private static readonly DateTime _unixEpoch = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
+#endif
+
+        private readonly int _secondsBetweenLogs;
+        private readonly ConcurrentDictionary<int, LogRateBucketInfo> _buckets
+            = new ConcurrentDictionary<int, LogRateBucketInfo>();
+
+        public LogRateLimiter(int secondsBetweenLogs)
+        {
+            if (secondsBetweenLogs < 1)
+            {
+                throw new ArgumentException("Must have positive number of seconds between logs", nameof(secondsBetweenLogs));
+            }
+
+            _secondsBetweenLogs = secondsBetweenLogs;
+        }
+
+        /// <inheritdoc/>
+        public bool ShouldLog(string filePath, int lineNumber, out uint skipCount)
+        {
+            // RFC says we should take log context, level, filename and lineNumber into account
+            // but we don't currently set the log context name in IDatadogLogger. FilePath and
+            // lineNumber should generally sufficient to uniquely identify the log given our API anyway
+            var key = HashCode.Combine(filePath, lineNumber);
```

Comment:
I believe you should use a filePath/lineNumber struct (where you override the Equals/GetHashCode logic) as key, so that the ConcurrentDictionary can take care of hash collisions. Right now it's entirely possible that two different logs exclude each other because of hash collision

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/1153/commits/05de3984613bd001f05f2c5f22b34db3ccb4d2e5